hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/nio/NioEventloop.java

```diff
@@ -71,6 +71,9 @@ protected void run() throws Exception {
                     keyCount = selector0.selectNow();
                 }
                 wakeupNeeded0.set(false);
+                // we need to update the clock because we could have been blocked for quite
+                // some time and clock could be very much out of sync.
+                nanoClock0.update();
```

Comment:
When we reach here, it doesn't mea that we always did a blocking operation. For example, if `externalTaskQueue0` isn't empty, we do a non-blocking call `selectNow()` yet still we update the cached time here.

Reply:
That is fine. It is still correct.

Reply:
I made a small adjustment to reduce the need for clock.update.

Reply:
We can even reduce further, but since you already knew that I'll resolve this.